io/disk/softwareraid.py

```diff
@@ -0,0 +1,112 @@
+#!/usr/bin/env python
+
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+#
+# Copyright: 2016 IBM
+# Author: Venkat Rao B <vrbagal1@linux.vnet.ibm.com>
+
+
+"""
+RAID devices are virtual devices created from two or more real block devices.
+This allows multiple devices (typically disk drives or partitions thereof) to
+be combined into a single device to hold (for example) a single filesystem.
+Some RAID levels include redundancy and so can survive some degree of device
+failure.
+"""
+
+import re
+from avocado import Test
+from avocado.utils.software_manager import SoftwareManager
+from avocado.utils import process
+from avocado import main
+
+
+class SoftwareRaid(Test):
+
+    """
+    This scripts create, assembles and stops md device, using mdadm tool
+    """
+
+    def setUp(self):
+
+        """
+        Checking if the required packages are installed,
+        if not found packages will be installed
+        """
+
+        smm = SoftwareManager()
+        if not smm.check_installed("mdadm"):
+            self.skip("Mdadm must be installed before continuing the test")
+            if SoftwareManager().install("mdadm") is False:
+                self.skip("Unable to install mdadm")
```

Comment:
here seems me repetition of code but  No strong opinion about it. rather than  use of  multiple code for package check it can be 
`if not sm.check_installed('mdadm') and not  sm.install('mdadm'):`  
so please consider it